source/module_base/complexarray.h

```diff
@@ -83,23 +89,37 @@ class ComplexArray
 ComplexArray operator*(double r, const ComplexArray &cd);
 ComplexArray operator*(std::complex < double> c, const ComplexArray &cd);
 
+//Sum of absolute squares of all elements in cd
 double abs2(const ComplexArray &cd);
 
+
 void
 add_scale_abs2(const std::complex < double> &c, const ComplexArray & in,
                ComplexArray &out);
 
+// Take "dot-product" of two ComplexArray:  sum of cd1(conjugate)[i] * cd2[i]
 std::complex < double> dot(const ComplexArray &cd1, const ComplexArray &cd2);
+
+/* Does cd2 += r * cd1 */
 void scale_accumulate(double r, const ComplexArray &cd1, ComplexArray &cd2);
+
+/* Does cd2 += c * cd1 */
 void scale_accumulate(std::complex < double> c, const ComplexArray &cd1, ComplexArray &cd2);
+
+/* Does cd3 = r1*cd1 + r2*cd2 */
 void scaled_sum(double r1, const ComplexArray &cd1,
                 double r2, const ComplexArray &cd2,
                 ComplexArray &cd3);
 
-void scaled_sum(double r1, ComplexArray &cd1,
-                double r2, ComplexArray &cd2,
-                ComplexArray &cd3);
+/* Does cd3 = c1*cd1 + c2*cd2 */
+void scaled_sum(std::complex < double> c1, const ComplexArray &cd1,
+           std::complex < double> c2, const ComplexArray &cd2,
+           ComplexArray &cd3);
+//void scaled_sum(double r1, ComplexArray &cd1,
```

Comment:
Why commented codes here?

Reply:
the commented function has been declared at line 110. This should be a mistake of previous author, he may like to declare the function at line 115 which is actually defined in .cpp file.

Reply:
> the commented function has been declared at line 110. This should be a mistake of previous author, he may like to declare the function at line 115 which is actually defined in .cpp file.

OK. If duplicated, just remove them.